docs/book/src/topics/addons.md

```diff
@@ -279,6 +279,22 @@ The following resources need to be updated when using `capi-quickstart.yaml` (th
                 .
     ```
 
+### Disable Azure network discovery (if using custom images)
+
+By default, Azure's images are coming with a `cloud-init` configuration that enables automatic network discovery mechanism that interferes with the Azure CNI.
```

Comment:
Please more clarity on what is custom image versus what is provided since the fix is baked into the default ones.

Reply:
Done!